src/core/shape_inference/include/search_sorted_shape_inference.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/op/search_sorted.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v15 {
+template <class TShape, class TRShape = result_shape_t<TShape>>
+std::vector<TRShape> shape_infer(const SearchSorted* op, const std::vector<TShape>& input_shapes) {
+    const auto& sorted_shape = input_shapes[0];
+    const auto& values_shape = input_shapes[1];
+    auto output_shape = values_shape;
+    TShape::merge_into(output_shape, sorted_shape);
```

Comment:
Assert the result e.g.:
```suggestion
    NODE_SHAPE_INFER_CHECK(op, TShape::merge_into(output_shape, sorted_shape), input_shapes, "Shapes are not compatible ....");
```
Is required when static shapes used (CPU).

And `merge_into` do same validation as [here](https://github.com/openvinotoolkit/openvino/pull/26904/files#diff-591173189ba331aa8f2ee9d6e6f3fa6fcae26dd755a5adeec3311ef107b7353aR34)